OSVROpenGL/app/src/main/jni/main.cpp

```diff
@@ -652,6 +626,12 @@ void stop() {
         delete gClientContext;
         gClientContext = nullptr;
     }
+
+    if(gServer) {
+        LOGI("[OSVR] Shutting down server...");
+        gServer->stop();
+        gServer = nullptr;
```

Comment:
I personally prefer a `.reset()` since it makes it clear you're using a smart pointer, but a comment saying that this will free the object is probably just as good/better.